### PR TITLE
4.6.3 prep, upstream propegation

### LINKED ISSUE #1324: pkg-config references unknown shared libraries
The lintian QA tool reported several `pkg-config-references-unknown-shared-library` issues for the Debian package build of netcdf-c 4.6.2.1:

 * `usr/lib/x86_64-linux-gnu/pkgconfig/netcdf.pc` `-lhdf5_hl` (line 11)
 * `usr/lib/x86_64-linux-gnu/pkgconfig/netcdf.pc` `-lhdf5` (line 11)
 * `usr/lib/x86_64-linux-gnu/pkgconfig/netcdf.pc` `-lpthread` (line 11)
 * `usr/lib/x86_64-linux-gnu/pkgconfig/netcdf.pc` `-lsz` (line 11)
 * `usr/lib/x86_64-linux-gnu/pkgconfig/netcdf.pc` `-lz` (line 11)
 * `usr/lib/x86_64-linux-gnu/pkgconfig/netcdf.pc` `-ldl` (line 11)
 * `usr/lib/x86_64-linux-gnu/pkgconfig/netcdf.pc` `-lm` (line 11)
 * `usr/lib/x86_64-linux-gnu/pkgconfig/netcdf.pc` `-lcurl` (line 11)

The lintian tag description is:
> The specified pkg-config(1) file references a shared object via, for
> example, Libs: -lfoo but this package appears to not ship the
> associated "libfoo.so" shared library.
> 
> This will result in a linker error and was likely caused by a missing
> installation step.
> 
> Please ensure that your package ships the corresponding libfoo.so
> shared object file.
> 
> Refer to the pkg-config(1) manual page and
> https://bugs.debian.org/919180 for details.

`netcdf.pc` contains:
```
prefix=/usr
exec_prefix=/usr
libdir=/usr/lib/x86_64-linux-gnu
includedir=/usr/include
ccompiler=/usr/bin/cc

Name: netCDF
Description: NetCDF Client Library for C
URL: http://www.unidata.ucar.edu/netcdf
Version: 4.6.2.1
Libs: -L${libdir} -L/usr/lib/x86_64-linux-gnu/hdf5/serial -lnetcdf -lhdf5_hl -lhdf5 -lpthread -lsz -lz -ldl -lm -lcurl
Cflags: -I${includedir} -I/usr/include/hdf5/serial
```

And `pkg-config(1)` documents:
> Libs:  This line should give the link flags specific to your package.  Don't add any flags for required packages; pkg-config will add those automatically.

`netcdf.pc.in` should probably not use `@NC_LIBS@` but only `-lnetcdf`.

### Environment Information

> Feel free to skip this if the issue is related to documentation, a feature request, or general discussion.

* What platform are you using? (please provide specific distribution/version in summary)
  * [x] Linux
  * [ ] Windows
  * [ ] OSX
  * [ ] Other
  * [ ] NA
* 32 and/or 64 bit?
  * [ ] 32-bit
  * [x] 64-bit
* What build system are you using?
  * [ ] autotools (`configure`)
  * [x] cmake
* Can you provide a sample netCDF file or `C` code to recreate the issue?
  * [ ] Yes (please attach to this issue, thank you!)
  * [x] No
  * [ ] Not at this time

### PR DESCRIPTION
**Pull request by WardF**
* Fixes #1324 

Development branch to handle those changes in 4.6.3 that need to be merged back upstream.

### CODE REVIEW

File: plugins/H5Zbzip2.c [hunk 1]
```diff
@@ -7,6 +7,8 @@
 /* Older versions of the hdf library may define H5PL_type_t here */
 #include <H5PLextern.h>
 
+#include "config.h"
```
Comment (edhartnett) [inline plugins/H5Zbzip2.c]: You need to put config.h before any other includes. Some of the settings in config.h may well impact other includes, so config.h has to come first.

**Comment by WardF** [comment]
Not immediately available. I could create and attach one in a bit, but working on updating my internal windows package creating tool at the moment.

**Comment by WardF** [comment]
If you have access to docker, you can run the following command in the meantime:

    $ docker run -e OLDVER=[version] -e NEWVER=[version] -v [local output directory]:/output unidata/ncabi

I invoke it as follows:

    $ mkdir tmp && cd tmp $$ docker run -e OLDVER=v4.6.2 -e NEWVER=v4.6.3-prep.wif -v $(pwd):/output unidata/ncabi

**Comment by WardF** [comment]
* [compat_reports.zip](https://github.com/Unidata/netcdf-c/files/2917169/compat_reports.zip)

**Comment by Dave-Allured** [comment]
Thanks Ward for the compat report.  Now I see a good bunch of ABI backward compatibility issues.  Therefore I concur with your SO version bump from 14.2.1 to 15.0.0.  "Case closed."

**Pull request merged**